src/main/scala/firrtl/passes/ReplaceMemMacros.scala

```diff
@@ -95,75 +40,86 @@ class ReplaceMemMacros(writer: ConfWriter) extends Pass {
+  def maskBits(mask: WSubField, dataType: Type, fillMask: Boolean) =
+    if (fillMask) toBitMask(mask, dataType) else toBits(mask)
+
+  def adaptReader(wrapperPort: WRef, bbPort: WSubField) =
+    defaultConnects(wrapperPort, bbPort) :+
+    fromBits(createSubField(wrapperPort, "data"), createSubField(bbPort, "data"))
+
+  def adaptWriter(wrapperPort: WRef, bbPort: WSubField, hasMask: Boolean, fillMask: Boolean) = {
+    val wrapperData = toBits(createSubField(wrapperPort, "data"))
+    val defaultSeq = defaultConnects(wrapperPort, bbPort) :+
+      Connect(NoInfo, createSubField(bbPort, "data"), wrapperData)
+    hasMask match {
+      case false => defaultSeq
+      case true => defaultSeq :+ Connect(
         NoInfo,
-        WSubField(bbPort, "mask", bbMask, UNKNOWNGENDER),
-        rhs
+        createSubField(bbPort, "mask"),
+        maskBits(createSubField(wrapperPort, "mask"), wrapperData.tpe, fillMask)
       )
     }
-    else defaultSeq
   }
 
-  def adaptReadWriter(wrapperPort: Expression, wrapperMem: DefMemory, bbPort: Expression, bbMem: DefMemory) = {
-    val defaultSeq = Seq(
-      connectFields(bbPort, "addr", wrapperPort, "addr"),
-      connectFields(bbPort, "en", wrapperPort, "en"),
-      connectFields(bbPort, "clk", wrapperPort, "clk"),
-      connectFields(bbPort, "wmode", wrapperPort, "wmode"),
-      Connect(
-        NoInfo,
-        WSubField(bbPort, "wdata", bbMem.dataType, UNKNOWNGENDER),
-        toBits(WSubField(wrapperPort, "wdata", wrapperMem.dataType, UNKNOWNGENDER))
-      ),
-      fromBits(
-        WSubField(wrapperPort, "rdata", wrapperMem.dataType, UNKNOWNGENDER),
-        WSubField(bbPort, "rdata", bbMem.dataType, UNKNOWNGENDER)
-      )
-    )
-    if (containsInfo(wrapperMem.info, "maskGran")) {
-      val wrapperMask = createMask(wrapperMem.dataType)
-      val fillWMask = getFillWMask(wrapperMem)
-      val bbMask = if (fillWMask) flattenType(wrapperMem.dataType) else flattenType(wrapperMask)
-      val rhs = {
-        if (fillWMask) toBitMask(WSubField(wrapperPort, "wmask", wrapperMask, UNKNOWNGENDER), wrapperMem.dataType)
-        else toBits(WSubField(wrapperPort, "wmask", wrapperMask, UNKNOWNGENDER))
-      }
-      defaultSeq :+ Connect(
+  def adaptReadWriter(wrapperPort: WRef, bbPort: WSubField, hasMask: Boolean, fillMask: Boolean) = {
+    val wrapperWData = toBits(createSubField(wrapperPort, "wdata"))
+    val defaultSeq = defaultConnects(wrapperPort, bbPort) ++ Seq(
+      fromBits(createSubField(wrapperPort, "rdata"), createSubField(bbPort, "rdata")),
+      connectFields(bbPort, "wmode", wrapperPort, "wmode"), 
+      Connect(NoInfo, createSubField(bbPort, "wdata"), wrapperWData))
+    hasMask match {
+      case false => defaultSeq
+      case true => defaultSeq :+ Connect(
         NoInfo,
-        WSubField(bbPort, "wmask", bbMask, UNKNOWNGENDER),
-        rhs
+        createSubField(bbPort, "wmask"),
+        maskBits(createSubField(wrapperPort, "wmask"), wrapperWData.tpe, fillMask)
```

Comment:
My guess here is rocket-chip actually don't need this, so it worked with this broken code...